src/cephadm/cephadm.py

```diff
@@ -7732,11 +7754,12 @@ def cmd_is_available(cmd: str) -> bool:
         key = f.read().strip()
     new_key = authorize_ssh_key(key, ctx.ssh_user)
     ssh_cfg_file_arg = ['-F', pathify(ctx.ssh_config.name)] if ctx.ssh_config else []
+    load_sudo_password(ctx)
     _, _, code = call(ctx, ['ssh', '-o StrictHostKeyChecking=no',
                             *ssh_cfg_file_arg, '-i', ssh_priv_key_path,
                             '-o PasswordAuthentication=no',
                             f'{ctx.ssh_user}@{get_hostname()}',
-                            'sudo echo'])
+                            f'export SUDO_ASKPASS=echo {ctx.sudo_password}; sudo --askpass echo' if ctx.sudo_password else 'sudo echo'])
```

Comment:
I'm not sure what this is supposed to be doing. I do see it echoing the password as part of a command line and that goes back to my original concern with putting the password on the command line. If you remember my earlier recommendation was to write to a named pipe (over the ssh connection).

Reply:
just remotely put a script with pass to avoid exposure of credentials on the command line. Will look to add some tests for the changes if all above looks good.